compiler/optimizer/OMRCFGSimplifier.cpp

```diff
@@ -1166,6 +1172,8 @@ bool OMR::CFGSimplifier::simplifyBooleanStore(bool needToDuplicateTree)
       }
    TR::Node *value1 = store1->getChild(valueIndex);
    TR::Node *value2 = store2->getChild(valueIndex);
+   if (value1->isInternalPointer() || value2->isInternalPointer())
```

Comment:
Could this moved up with the opcode checks etc and/or adding some tracing to know we passed this test?